integrations/adobe-analytics/lib/index.js

```diff
@@ -234,56 +419,74 @@ AdobeAnalytics.prototype.loaded = function() {
   var pageName = page.fullName();
-  // TODO: for nameless analytics.page(), pageName is `undefined`
-  // Should we be setting or sending something else here?
-  // When window.s.pageName is not set, AA falls back on url which is bad.
-  // Not sure what happens when it is sent as `undefined` (not string)
-  // Either way, any change here would be breaking
-  window.s.pageName = pageName;
-  window.s.referrer = page.referrer();
-
-  // Visitor ID aka AA's concept of `userId`.
-  // This is not using `update()` so once it is set, it will be sent
-  // with `.track()` calls as well.
-  // visitorId is not supported for timestamped hits
-  // https://marketing.adobe.com/resources/help/en_US/sc/implement/timestamps-overview.html
-  if (!this.options.disableVisitorId) {
-    var userId = this.analytics.user().id();
-    if (userId) {
-      if (this.options.timestampOption === 'disabled')
-        window.s.visitorID = userId;
-      if (
-        this.options.timestampOption === 'hybrid' &&
-        this.options.preferVisitorId
-      )
-        window.s.visitorID = userId;
+
+  if (this.options.chromecastMode) {
+    //var props = extractProperties(page, this.options);
+    if (this.analytics && this.analytics.user()) {
+      var userId = this.analytics.user().id();
+      window.ADBMobile.config.setUserIdentifier(userId);
+
+    } else if (this.analytics.user().anonymousId()) {
+      var anonymousId = this.analytics.user().anonymousId();
+      window.ADBMobile.config.setUserIdentifier(anonymousId);
+    }
+
+
+
+    window.ADBMobile.analytics.trackState(pageName);
+
+  } else {
+    // TODO: for nameless analytics.page(), pageName is `undefined`
+    // Should we be setting or sending something else here?
+    // When window.s.pageName is not set, AA falls back on url which is bad.
+    // Not sure what happens when it is sent as `undefined` (not string)
+    // Either way, any change here would be breaking
+    window.s.pageName = pageName;
+    window.s.referrer = page.referrer();
+
+    // Visitor ID aka AA's concept of `userId`.
+    // This is not using `update()` so once it is set, it will be sent
+    // with `.track()` calls as well.
+    // visitorId is not supported for timestamped hits
+    // https://marketing.adobe.com/resources/help/en_US/sc/implement/timestamps-overview.html
+    if (!this.options.disableVisitorId) {
+      var userId = this.analytics.user().id();
+      if (userId) {
+        if (this.options.timestampOption === 'disabled')
+          window.s.visitorID = userId;
+        if (
+          this.options.timestampOption === 'hybrid' &&
+          this.options.preferVisitorId
+        )
+          window.s.visitorID = userId;
+      }
     }
-  }
 
-  // Attach some variables on the `window.s` to be sent with the call
-  update(pageName, 'events');
-  updateCommonVariables(page, this.options);
+    // Attach some variables on the `window.s` to be sent with the call
+    update(pageName, 'events');
```

Comment:
more diffs from style changes -- we should avoid creating this diff to make PR more focused on functional changes.

Reply:
i will fix this along with any other style changes!

Reply:
I think this specific change will have a style diff since it needs to be wrapped in an `else` statement whereas before it wasn't, so it's indented further in now